### ISSUE TITLE
ImagePicker.RESULT_ERROR not valid when combined with InlineActivityResult

### ISSUE THREAD

**Issue by angelolloqui** [Dhaval2404/ImagePicker#108]
I am using this library together with InlineActivityResult dependency.

However, when the user rejects permissions, I see the code in ImagePicker calling the `setError` handler into `ImagePickerActivity`, but that error is never returned to the lambda provided with the `imagePicker.start { resultCode, data ->`

Digging into the code, I see this part in `ImagePickerActivity`:

```
    fun setError(message: String) {
        val intent = Intent()
        intent.putExtra(ImagePicker.EXTRA_ERROR, message)
        setResult(ImagePicker.RESULT_ERROR, intent)
        finish()
    }
```

and this in the `InlineActivityResult.java`:

```
    private void onReceivedActivityResult(int requestCode, int resultCode, @Nullable final Intent data) {
        final Result result = new Result(this, requestCode, resultCode, data);
        if (resultCode == Activity.RESULT_OK) {
            for (SuccessCallback callback : successCallbacks) {
                callback.onSuccess(result);
            }
            for (ActivityResultListener listener : responseListeners) {
                listener.onSuccess(result);
            }
        } else if (resultCode == Activity.RESULT_CANCELED) {
            for (FailCallback callback : failCallbacks) {
                callback.onFailed(result);
            }
            for (ActivityResultListener listener : responseListeners) {
                listener.onFailed(result);
            }
        }
    }
```

So, basically, it only fires the callbacks when the result code is either `RESULT_OK` or `RESULT_CANCELED`, but it does not fire for other custom codes.

Since the library uses a custom error code for the errors, it is not being propagated. 

Could `ImagePicker` use regular error codes? it makes the whole usage of InlineActivityResult useless, since we lose one of the most important scenarios, the permission rejected one.

**Comment by Dhaval2404** [comment]
@angelolloqui Thank you for the detailed analysis! I will soon fix the issue!

**Comment by Dhaval2404** [comment]
Closing this task as ImagePicker no longer depends on InlineActivityResult
